jupyter_server/gateway/connections.py

```diff
@@ -146,6 +146,9 @@ def handle_outgoing_message(self, incoming_msg: str, *args: Any) -> None:
     def handle_incoming_message(self, message: str) -> None:
         """Send message to gateway server."""
         if self.ws is None and self.ws_future is not None:
+            if self.ws_future.done() and isinstance(self.ws_future.exception(), Exception):
+                self.log.warning(f"Exception connect to websocket {self.ws_future.exception()}")
```

Comment:
We should only log the Exception here if an exception in `ws_future` wouldn't already be logged, but it [is](https://github.com/jupyter-server/jupyter_server/blob/d85e4810df12c43221dd81645f0bb73f6b648977/jupyter_server/gateway/connections.py#L72). How about a simpler:

```suggestion
            if self.ws_future.done() and self.ws_future.exception() is not None:
                self.log.warning("Ignoring message on failed connection to kernel %s", self.kernel_id)
```

Reply:
ok